Client/Pages/Home.razor

```diff
@@ -6,19 +6,52 @@
 @using Microsoft.AspNetCore.Components
 @using System.Net.Http
 @using System.Net.Http.Json
+@using BlazorApp.Client.Services
 
 @inject Blazored.LocalStorage.ILocalStorageService localStorage
 @inject NavigationManager Navigation
 @inject HttpClient Http
+@inject IAuthService AuthService
 
 <PageTitle>Home</PageTitle>
 
+@if (currentUser != null)
+{
+    <div class="welcome-message mb-3">
+        <h3>Welcome, @(currentUser.FirstName ?? currentUser.Name ?? "User")!</h3>
```

Comment:
You'll need to check if these strings are empty as well as null. Also include `UserDetails` (which is the user's email address)